extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -325,16 +342,26 @@ export class ProjectsController {
 			return; // user cancelled
 		}
 
-		// TODO: file already exists?
-
 		const newFileText = this.macroExpansion(itemType.templateScript, { 'OBJECT_NAME': itemObjectName });
 		const relativeFilePath = path.join(relativePath, itemObjectName + constants.sqlFileExtension);
 
-		const newEntry = await project.addScriptItem(relativeFilePath, newFileText);
+		try {
+			// check if file already exists
+			const absoluteFilePath = path.join(project.projectFolderPath, relativeFilePath);
+			const fileExists = await utils.exists(absoluteFilePath);
 
-		this.apiWrapper.executeCommand('vscode.open', newEntry.fsUri);
+			if (fileExists) {
+				throw new Error(constants.fileAlreadyExists(path.parse(absoluteFilePath).name));
+			}
 
-		this.refreshProjectsTree();
+			const newEntry = await project.addScriptItem(relativeFilePath, newFileText);
+
+			this.apiWrapper.executeCommand('vscode.open', newEntry.fsUri);
+
+			this.refreshProjectsTree();
+		} catch (err) {
+			this.apiWrapper.showErrorMessage(utils.getErrorMessage(err));
```

Comment:
should the showErrorMessage ones be awaited? the vscode api is thenable...

@aaomidi  any suggestion here?

Reply:
No they shouldn't. The await only resolves once the dialog is closed. The promise is kinda being used as an event.